src/pydisagg/models.py

```diff
@@ -29,6 +30,41 @@ def __init__(
             error_inflation=error_inflation,
             beta_standard_error=beta_standard_error
         )
+    
+    def fit_beta(
+        self,
+        bucket_populations: NDArray,
+        observed_total: float,
+        observed_total_se: Optional[float] = None,
+        rate_pattern: Optional[NDArray] = None,
+        lower_guess: Optional[float] = -50,
+        upper_guess: Optional[float] = 50,
+        verbose: Optional[int] = 0
+    )->None:
+        """
+        Custom fit_beta for this model, as we can do it without rootfinding. 
+        """
+        _ = self.pull_set_rate_pattern(rate_pattern)
```

Comment:
Hi @AHsu98! I think the function `pull_set_rate_pattern` is a little confusing. Usually the setter and getter are separate. I recommend using `property` to create a setter and getter for `rate_pattern`:
```python
class DisaggModel:
    def __init__(self, ..., rate_pattern: Optional[NDArray] = None, ...):
        self._rate_pattern = rate_pattern

    @property
    def rate_pattern(self) -> NDArray:
        if self._rate_pattern is None:
            raise ValueError("No Rate Pattern Available")
        return self._rate_pattern

    @rate_pattern.setter
    def rate_pattern(self, rate_pattern: Optional[NDArray]):
        if rate_pattern is not None:
            self._rate_pattern = rate_pattern

    def fit_beta(self, ...):
        self.rate_pattern = rate_pattern
```
For more detail on setter and getter, please check
https://realpython.com/python-getter-setter/#using-properties-instead-of-getters-and-setters-the-python-way

Reply:
Approve for now, will handle this in issue #22